data_model/src/asset.rs

```diff
@@ -20,9 +20,71 @@ use crate::{
     Identifiable, Name, ParseError, Registered, TryAsMut, TryAsRef, Value,
 };
 
-/// [`AssetsMap`] provides an API to work with collection of key ([`Id`]) - value
+#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+/// Key of [`AssetsMap`]
+pub struct AssetsMapKey {
+    /// [`Id`] of inserted [`Asset`]
+    pub asset_id: <Asset as Identifiable>::Id,
+    /// Insertion timestamp
+    pub timestamp: i64,
```

Comment:
I don't think you don't need a timestamp. Just one `usize` int tracking the insertion order

Reply:
also have you considered [this](https://github.com/bluss/indexmap). Might be an overkill for this purpose only

Reply:
> Preserves insertion order as long as you don't call .remove().

I don't think that it fits our requirements.